src/DotNetLightning.Core/Serialization/Msgs/Msgs.fs

```diff
@@ -597,9 +608,18 @@ with
             this.DelayedPaymentBasepoint <- ls.ReadDelayedPaymentBasepoint()
             this.HTLCBasepoint <- ls.ReadHtlcBasepoint()
             this.FirstPerCommitmentPoint <- ls.ReadPerCommitmentPoint()
+            let tlvs =
+                ls.ReadTLVStream()
+                |> Array.map(AcceptChannelTLV.FromGenericTLV)
             this.ShutdownScriptPubKey <-
-                if (ls.Position = ls.Length) then None else
-                ls.ReadWithLen() |> Script |> Some
+                Seq.choose
+                    (function
+                        | AcceptChannelTLV.UpfrontShutdownScript script -> Some script
+                        | _ -> None
+                    )
+                    tlvs
+                |> Seq.tryExactlyOne
```

Comment:
I don't think discarding `Unknown` fields like this is good idea.
The fact that "There is an Unknown TLV field sent from other party" is itself valuable and application might want to tell the user about it. Or it may want to pass the data to another application as-is.

I suggest to have a field `TLVs: AcceptChannelTLV[]` in `AcceptChannelMsg`  and add new property
```fs
this.ShutdownScriptPubKey: Script option =
    this.TLVs |> Seq.choose  // ...
```

Reply:
@joemphilips updated